[Static Analyzer CI] Rename scripts and processes from smart pointer to safer CPP

briannafan opened the pull request:
#### bc1c19d5d9028c4bed727da393f371de3997f17e
<pre>
[Static Analyzer CI] Rename scripts and processes from smart pointer to safer CPP
<a href="https://bugs.webkit.org/show_bug.cgi?id=280729">https://bugs.webkit.org/show_bug.cgi?id=280729</a>
<a href="https://rdar.apple.com/137096120">rdar://137096120</a>

Reviewed by Ryosuke Niwa and Ryan Haddad.

Reflect the nature of our static analysis better by changing references to smart pointer
to safer cpp or static analysis.

Also updated build summary for build-webkit-org DisplaySaferCPPResults to be more accurate
and useful based on the results. Added unittests for DisplaySaferCPPResults.

* Tools/Scripts/update-safer-cpp-expectations:
    Renamed from Tools/Scripts/smart-pointer-tool.
    Allow the user to pass in any filename if arguments are specified correctly.
(parser): Add arguments for adding and removing failures, specifying project name.
(modify_expectations_for_checker):
(update_expectations):
(is_expected_file):
(main):

* Tools/CISupport/build-webkit-org/config.json:
* Tools/CISupport/build-webkit-org/factories.py:
(SaferCPPStaticAnalyzerFactory):
(SaferCPPStaticAnalyzerFactory.__init__):
* Tools/CISupport/build-webkit-org/factories_unittest.py:
(TestExpectedBuildSteps):
* Tools/CISupport/build-webkit-org/steps.py:
(ScanBuild):
(ScanBuild.run):
(DisplaySaferCPPResults):
(UpdateSaferCPPBaseline):
* Tools/CISupport/build-webkit-org/steps_unittest.py:
* Tools/CISupport/ews-build/config.json:
* Tools/CISupport/ews-build/events.py:
(Events):
* Tools/CISupport/ews-build/factories.py:
(SaferCPPStaticAnalyzerFactory):
(SaferCPPStaticAnalyzerFactory.__init__):
* Tools/CISupport/ews-build/factories_unittest.py:
(TestExpectedBuildSteps):
* Tools/CISupport/ews-build/loadConfig.py:
* Tools/CISupport/ews-build/steps.py:
(UpdatePullRequest.hideStepIf):
(ScanBuild):
(ScanBuildWithoutChange):
(FindUnexpectedStaticAnalyzerResults.run):
(DisplaySaferCPPResults):
(DisplaySaferCPPResults.getResultSummary):
* Tools/CISupport/ews-build/steps_unittest.py:

Canonical link: <a href="https://commits.webkit.org/284596@main">https://commits.webkit.org/284596@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/39c5df3f70ec91a419b2ed0ea7d4dec211b0884d

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/69945 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/49346 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/22698 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/74030 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/21103 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/72062 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/57146 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/20954 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/55507 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/13990 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/73011 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/44954 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/60328 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/35987 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/41619 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/17756 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/19480 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/63542 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/18107 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/75745 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/14170 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/17332 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/63202 "Passed tests") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/69623 "Passed tests") | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/14206 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/60403 "Passed tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/63138 "Build is in progress. Recent messages:Printed configuration; Running apply-patch; Checked out pull request; run-api-tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/11157 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/4765 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/10675 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/45149 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/46223 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/47494 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/45964 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

ryanhaddad reviewed (approved):
nit is that it would have been easier to review as multiple PRs since its not just a rename like the title suggests, but r=me assuming tests on the dev instance look good.

briannafan committed:
[Static Analyzer CI] Rename scripts and processes from smart pointer to safer CPP
https://bugs.webkit.org/show_bug.cgi?id=280729
rdar://137096120

Reviewed by Ryosuke Niwa and Ryan Haddad.

Reflect the nature of our static analysis better by changing references to smart pointer
to safer cpp or static analysis.

Also updated build summary for build-webkit-org DisplaySaferCPPResults to be more accurate
and useful based on the results. Added unittests for DisplaySaferCPPResults.

* Tools/Scripts/update-safer-cpp-expectations:
    Renamed from Tools/Scripts/smart-pointer-tool.
    Allow the user to pass in any filename if arguments are specified correctly.
(parser): Add arguments for adding and removing failures, specifying project name.
(modify_expectations_for_checker):
(update_expectations):
(is_expected_file):
(main):

* Tools/CISupport/build-webkit-org/config.json:
* Tools/CISupport/build-webkit-org/factories.py:
(SaferCPPStaticAnalyzerFactory):
(SaferCPPStaticAnalyzerFactory.__init__):
* Tools/CISupport/build-webkit-org/factories_unittest.py:
(TestExpectedBuildSteps):
* Tools/CISupport/build-webkit-org/steps.py:
(ScanBuild):
(ScanBuild.run):
(DisplaySaferCPPResults):
(UpdateSaferCPPBaseline):
* Tools/CISupport/build-webkit-org/steps_unittest.py:
* Tools/CISupport/ews-build/config.json:
* Tools/CISupport/ews-build/events.py:
(Events):
* Tools/CISupport/ews-build/factories.py:
(SaferCPPStaticAnalyzerFactory):
(SaferCPPStaticAnalyzerFactory.__init__):
* Tools/CISupport/ews-build/factories_unittest.py:
(TestExpectedBuildSteps):
* Tools/CISupport/ews-build/loadConfig.py:
* Tools/CISupport/ews-build/steps.py:
(UpdatePullRequest.hideStepIf):
(ScanBuild):
(ScanBuildWithoutChange):
(FindUnexpectedStaticAnalyzerResults.run):
(DisplaySaferCPPResults):
(DisplaySaferCPPResults.getResultSummary):
* Tools/CISupport/ews-build/steps_unittest.py:

Canonical link: https://commits.webkit.org/284596@main
